pkg/operator/controller/controller_dns_configmap.go

```diff
@@ -115,18 +142,59 @@ func (r *reconciler) currentDNSConfigMap(dns *operatorv1.DNS) (bool, *corev1.Con
 	return true, current, nil
 }
 
-func desiredDNSConfigMap(dns *operatorv1.DNS, clusterDomain string) (*corev1.ConfigMap, error) {
+func desiredDNSConfigMap(dns *operatorv1.DNS, clusterDomain string, caBundleRevisionMap map[string]string) (*corev1.ConfigMap, error) {
 	if len(clusterDomain) == 0 {
 		clusterDomain = "cluster.local"
 	}
 
+	for _, server := range dns.Spec.Servers {
+		transport := server.ForwardPlugin.TransportConfig.Transport
+		tls := server.ForwardPlugin.TransportConfig.TLS
+		if transport == operatorv1.TLSTransport {
+			// tls cannot be configured without a ServerName
+			if tls == nil || tls.ServerName == "" {
+				return nil, errTransportTLSConfiguredWithoutServerName
+			}
+			// tls can only be configured for ip addresses
+			for _, upstream := range server.ForwardPlugin.Upstreams {
+				if _, _, err := net.SplitHostPort(upstream); err != nil {
+					if res := net.ParseIP(upstream); res == nil {
+						return nil, errTransportTLSConfiguredForSysResConf
```

Comment:
Does the fact that it fails to `ParseIP` mean that it's `operatorv1.SystemResolveConfType`? Just curious if this error message is always describing the right error.

Reply:
You're right, that might lead to showing the wrong error. I changed the error to `errTransportTLSConfiguredForNonIP` to cover both cases.